packages/front-end/components/MetricAnalysis/MetricAnalysis.tsx

```diff
@@ -0,0 +1,784 @@
+import React, { FC, useMemo, useState } from "react";
+import { useForm } from "react-hook-form";
+import {
+  FaDatabase,
+  FaExclamationTriangle,
+  FaQuestionCircle,
+} from "react-icons/fa";
+import {
+  CreateMetricAnalysisProps,
+  MetricAnalysisInterface,
+  MetricAnalysisPopulationType,
+  MetricAnalysisResult,
+  MetricAnalysisSettings,
+} from "back-end/types/metric-analysis";
+import { FactMetricInterface } from "back-end/types/fact-table";
+import { DataSourceInterfaceWithParams } from "back-end/types/datasource";
+import clsx from "clsx";
+import RunQueriesButton, {
+  getQueryStatus,
+} from "@/components/Queries/RunQueriesButton";
+import useApi from "@/hooks/useApi";
+import Toggle from "@/components/Forms/Toggle";
+import DateGraph from "@/components/Metrics/DateGraph";
+import HistogramGraph from "@/components/MetricAnalysis/HistogramGraph";
+import IdentifierChooser from "@/components/MetricAnalysis/IdentifierChooser";
+import PopulationChooser from "@/components/MetricAnalysis/PopulationChooser";
+import Field from "@/components/Forms/Field";
+import SelectField from "@/components/Forms/SelectField";
+import {
+  formatNumber,
+  getColumnRefFormatter,
+  getExperimentMetricFormatter,
+} from "@/services/metrics";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import { useLocalStorage } from "@/hooks/useLocalStorage";
+import Tooltip from "@/components/Tooltip/Tooltip";
+import { useAuth } from "@/services/auth";
+import QueriesLastRun from "@/components/Queries/QueriesLastRun";
+import MetricAnalysisMoreMenu from "@/components/MetricAnalysis/MetricAnalysisMoreMenu";
+import ViewAsyncQueriesButton from "@/components/Queries/ViewAsyncQueriesButton";
+import track from "@/services/track";
+
+function MetricAnalysisOverview({
+  name,
+  metricType,
+  userIdType,
+  result,
+  formatter,
+  numeratorFormatter,
+  denominatorFormatter,
+}: {
+  name: string;
+  metricType: string;
+  userIdType: string;
+  result: MetricAnalysisResult;
+  formatter: (value: number, options?: Intl.NumberFormatOptions) => string;
+  numeratorFormatter?: (
+    value: number,
+    options?: Intl.NumberFormatOptions
+  ) => string;
+  denominatorFormatter?: (
+    value: number,
+    options?: Intl.NumberFormatOptions
+  ) => string;
+}) {
+  let numeratorText: string;
+  let denominatorText: JSX.Element;
+  let numeratorValue: JSX.Element;
+  let denominatorValue: JSX.Element;
+  if (metricType === "ratio" && numeratorFormatter && denominatorFormatter) {
+    numeratorText = "Numerator: ";
+    denominatorText = <>{"Denominator: "}</>;
+    numeratorValue = <>{numeratorFormatter(result.numerator ?? 0)}</>;
+    denominatorValue = <>{denominatorFormatter(result.denominator ?? 0)}</>;
+  } else {
```

Comment:
you could just set your inital values to this then override if meets the if satement above

Reply:
i feel like none of them need to be JSX elements as well you can just set them all as string so no need for `= <>{bla}</>` you can just do `= bla`